packages/client/src/devices/InputMediaDeviceManagerState.ts

```diff
@@ -0,0 +1,94 @@
+import { BehaviorSubject, Observable, map } from 'rxjs';
+import { RxUtils } from '../store';
+
+export class InputMediaDeviceManagerState {
+  private mediaStreamSubject = new BehaviorSubject<MediaStream | undefined>(
+    undefined,
+  );
+  private selectedDeviceSubject = new BehaviorSubject<string | undefined>(
+    undefined,
+  );
+
+  /**
+   * An Observable that emits the current media stream, or `undefined` if the device is currently disabled.
+   *
+   */
+  mediaStream$: Observable<MediaStream | undefined>;
+
+  /**
+   * An Observable that emits the currently selected device
+   */
+  selectedDevice$: Observable<string | undefined>;
+
+  /**
+   * An Observable that emits the device status
+   */
+  // TODO: we should extend this list with loading and error status options
+  status$: Observable<'enabled' | 'disabled'>;
```

Comment:
Could status be infered from `mediaStream$`? Meaning, that we would not need `status$` state variable?

Reply:
We have the `status$` for parity with android, but status is computed from `mediaStream` https://github.com/GetStream/stream-video-js/blob/new-device-api/packages/client/src/devices/InputMediaDeviceManagerState.ts#L33

Reply:
Ok so this is about parity